### PR TITLE
MatchingDeclarationName now supports platofrm suffixes

### PR DESCRIPTION
**Pull request by lexa-diky**
This is one way of implementing #6401.

We can also consider removing default platform suffixes including 'common' which will break backward compatibility somewhat, but will make this rule much more flexible.

### CODE REVIEW

File: detekt-rules-naming/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/naming/MatchingDeclarationName.kt [hunk 1]
```diff
@@ -59,6 +59,9 @@ class MatchingDeclarationName(config: Config = Config.empty) : Rule(config) {
     @Configuration("name should only be checked if the file starts with a class or object")
     private val mustBeFirst: Boolean by config(true)
 
+    @Configuration("kotlin multiplatform target file suffixes")
+    private val multiplatformTargetFileSuffixes: List<String> by config(COMMON_KOTLIN_KMP_PLATFORM_TARGET_SUFFIXES)
```
Comment (lexa-diky) [inline detekt-rules-naming/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/naming/MatchingDeclarationName.kt:63]: maybe `multiplatformTargets`? `multiplatformTargetFileSuffixes` is super explicit, but also confusing whether consumer should put `.jvm.kt` or just `jvm`

Comment (BraisGabin) [reply detekt-rules-naming/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/naming/MatchingDeclarationName.kt:63]: Agree, multiplatformTarget is a better name.

File: detekt-rules-naming/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/naming/MatchingDeclarationName.kt [hunk 2]
```diff
@@ -89,4 +92,20 @@ class MatchingDeclarationName(config: Config = Config.empty) : Rule(config) {
...
+            "ios",
+            "android",
+            "js",
+            "jvm",
+            "native",
+            "iosArm64",
+            "iosX64",
+            "macosX64",
+            "mingwX64",
+            "linuxX64"
```
Comment (BraisGabin) [inline detekt-rules-naming/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/naming/MatchingDeclarationName.kt:108]: These are subtipes of the other ones, right? Would it have sense to remove them from the default configuration?


```suggestion
```

Comment (lexa-diky) [reply detekt-rules-naming/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/naming/MatchingDeclarationName.kt:108]: They are still pretty common, for example IJ has them in template. I suggest leaving them in

**Review by BraisGabin — APPROVED** [review]
Thanks!!

**Pull request merged**